solidity/contracts/BitcoinDepositor.sol

```diff
@@ -193,12 +222,12 @@ contract BitcoinDepositor is AbstractTBTCDepositor, Ownable2StepUpgradeable {
         if (address(_tbtcToken) == address(0)) {
             revert TbtcTokenZeroAddress();
         }
-        if (address(_stbtc) == address(0)) {
-            revert StbtcZeroAddress();
+        if (address(_acreVault) == address(0)) {
+            revert AcreVaultZeroAddress();
         }
 
         tbtcToken = IERC20(_tbtcToken);
-        stbtc = stBTC(_stbtc);
+        acreVault = IAcreVault(_acreVault);
```

Comment:
I... don't think it's going to work. If we do an upgrade, we need a `initializeV2(...) reinitializer(2)` function.

Reply:
Yes, this line in the initializer is for fresh deployment. For the upgrade we won't run it but execute the update function as you noticed.